src/spec/dsl.cr

```diff
@@ -130,8 +144,16 @@ module Spec
       return true if lines && lines.any? { |l| line == l || line <= l <= end_line }
     end
 
-    if spec_pattern || spec_line || locations
-      Spec::RootContext.matches?(description, spec_pattern, spec_line, locations)
+    if tags
+      tags = tags.to_set
+      return false if spec_anti_tags.try &.intersects?(tags)
+      return true if spec_tags.try &.intersects?(tags)
```

Comment:
I think this is wrong. It seems this is matching if the tags match, but what if the line doesn't match? I know it's maybe uncommon to specify a line and a tag, but... why is this check otuside of `Spec::RootContext.matches?`?

Reply:
Yeah, I agree...I have to do much more testing with respect to combinations of various filtering options